config/hubs/farallon.cluster.yaml

```diff
@@ -0,0 +1,164 @@
+            scheduler:
+              extraPodConfig:
+                nodeSelector:
+                  hub.jupyter.org/pool-name: dask-worker
+                tolerations:
+                  - key: "k8s.dask.org/dedicated"
+                    operator: "Equal"
+                    value: "worker"
+                    effect: "NoSchedule"
+                  - key: "k8s.dask.org_dedicated"
+                    operator: "Equal"
+                    value: "worker"
+                    effect: "NoSchedule"
+            worker:
+              extraPodConfig:
+                nodeSelector:
+                  hub.jupyter.org/pool-name: dask-worker
+                tolerations:
+                  - key: "k8s.dask.org/dedicated"
+                    operator: "Equal"
+                    value: "worker"
+                    effect: "NoSchedule"
+                  - key: "k8s.dask.org_dedicated"
+                    operator: "Equal"
+                    value: "worker"
+                    effect: "NoSchedule"
+
+            # TODO: figure out a replacement for userLimits.
+          extraConfig:
+            optionHandler: |
+              from dask_gateway_server.options import Options, Integer, Float, String
+              def cluster_options(user):
+                def option_handler(options):
+                    if ":" not in options.image:
+                        raise ValueError("When specifying an image you must also provide a tag")
+                    extra_annotations = {
+                        "hub.jupyter.org/username": user.name,
+                        "prometheus.io/scrape": "true",
+                        "prometheus.io/port": "8787",
+                    }
+                    extra_labels = {
+                        "hub.jupyter.org/username": user.name,
+                    }
+                    return {
+                        "worker_cores_limit": options.worker_cores,
+                        "worker_cores": min(options.worker_cores / 2, 1),
+                        "worker_memory": "%fG" % options.worker_memory,
+                        "image": options.image,
+                        "scheduler_extra_pod_annotations": extra_annotations,
+                        "worker_extra_pod_annotations": extra_annotations,
+                        "scheduler_extra_pod_labels": extra_labels,
+                        "worker_extra_pod_labels": extra_labels,
+                    }
+                return Options(
+                    Integer("worker_cores", 2, min=1, max=16, label="Worker Cores"),
+                    Float("worker_memory", 4, min=1, max=32, label="Worker Memory (GiB)"),
+                    String("image", default="pangeo/pangeo-notebook:latest", label="Image"),
+                    handler=option_handler,
+                )
+              c.Backend.cluster_options = cluster_options
+            idle: |
+              # timeout after 30 minutes of inactivity
+              c.KubeClusterConfig.idle_timeout = 1800
```

Comment:
OK, how did you build this config file?

Reply:
I see similar stuff in the pangeo-hub repo:
* https://github.com/2i2c-org/pangeo-hubs/blob/staging/hub/values.yaml
* https://github.com/2i2c-org/pangeo-hubs/tree/staging/deployments/farallon/config
Did you build it from those pieces?

Reply:
Yeah, I mostly copied that and hacked around it by testing on the current farallon-staging setup. The values in pangeo-hubs was further inspired by code in github.com/pangeo-data/pangeo-cloud-federation/